### PR TITLE
feature/relation-edit-mode

### PR DESCRIPTION
**Pull request by failipp**

### CODE REVIEW

File: client/src/modules/editor/modules/graph-editor/controls/GraphControls.ts [hunk 1]
```diff
@@ -161,42 +185,315 @@ export class GraphControls {
...
+        apiRelations.flat().forEach((rel) => {
+            relationMap.set(rel.relationId, rel);
+        });
+
+        // Prepare a set which will be filled with all DB relations which are already present in the graph
+        const alreadyPresentRelations = new Set<string>();
+
+        this.graphHandler.relations.forEach((relation, id) => {
+            if (!relation.uuid) {
+                console.log(relation.uuid, typeof relation.uuid);
```
Comment (stefan-rink) [inline client/src/modules/editor/modules/graph-editor/controls/GraphControls.ts:278]: ...

File: client/src/modules/editor/modules/graph-editor/store.ts [hunk 2]
```diff
@@ -95,13 +103,58 @@ export const graphEditor = {
...
+                    state.graphHandler.addCommand(command);
+                }
+            }
+        },
+
+        /**
+         * Add command for changed vertices to the undo redo stack
+         */
+        addChangeRelationVerticesCommand(state: GraphEditorState, verticesCommand: ChangeRelationVertexCommand): void {
+            if (state.graphHandler) state.graphHandler.addCommand(verticesCommand);
```
Comment (stefan-rink) [inline client/src/modules/editor/modules/graph-editor/store.ts:148]: `state.graphHandler?.addCommand(verticesCommand);`

File: client/src/modules/editor/i18n/de.json [hunk 3]
```diff
@@ -3,13 +3,19 @@
...
   "noDiagram": {
     "title": "Kein Diagram",
     "description": "Laden Sie erst ein Diagramm!"
   },
   "noNodes": {
     "description": "Keine Einträge gefunden!"
+  },
+  "relationModeToast": {
+    "title": "Beziehungsmodus is aktiv!",
+    "description": "Im Beziehungsmodus können keine keine neuen Nodes hinzugefügt werden"
```
Comment (luismoser99) [inline client/src/modules/editor/i18n/de.json:19]: Duplicate "keine"

File: client/src/models/Diagram.ts [hunk 4]
```diff
@@ -3,60 +3,7 @@ export class Diagram {
      * Mock graph
      */
     // language=JSON
```
Comment (luismoser99) [inline client/src/models/Diagram.ts:5]: Unneccessary comment remains

File: client/src/models/Diagram.ts [hunk 5]
```diff
@@ -3,60 +3,7 @@ export class Diagram {
      * Mock graph
```
Comment (luismoser99) [inline client/src/models/Diagram.ts:3]: Unneccessary comment

File: client/src/modules/editor/modules/graph-editor/controls/commands/CreateNodeCommand.ts [hunk 6]
```diff
@@ -10,6 +10,11 @@ export class CreateNodeCommand implements ICommand {
      */
     private diagElement?: dia.Element;
 
+    /**
+     * Array of relations that are already existing, needed for a redo
+     */
+    private existingRelations: Array<[Relation, dia.Element]> = [];
```
Comment (luismoser99) [inline client/src/modules/editor/modules/graph-editor/controls/commands/CreateNodeCommand.ts:16]: use new Array<> instead

**Review by luismoser99 — APPROVED** [review]
Looks good and works fine!

**Pull request merged**